### ISSUE TITLE
CUDA error: out of memory

### ISSUE THREAD

**Issue by bestz123** [soedinglab/MMseqs2#905]
## Expected Behavior
Hello, when I run mmseq_gpu on my own data set, this error is reported.  Is there any command restriction to prevent it from reporting such an error?

## Context
CUDA error: out of memory : /home/vsts/work/1/s/lib/libmarv/src/cudasw4.cuh, line 1276
Error: Prefilter died

## Your Environment
Include as many relevant details about the environment you experienced the bug in.
* Git commit used (The string after "MMseqs Version:" when you execute MMseqs without any parameters): 562a47f3a276721e40e63715474adf27747f1bfc
* Which MMseqs version was used (Statically-compiled, self-compiled, Homebrew, etc.): Statically-compiled
* Server specifications (especially CPU support for AVX2/SSE and amount of system memory): AMD 9654, 768G
* Operating system and version: ubuntu 20.04

**Comment by milot-mirdita** [comment]
What GPU are you using and what size is the target database? Can you list all commands you ran please?

**Comment by bestz123** [comment]
Hi @milot-mirdita，
The relevant information has been updated. I used A100 (80G) to perform msa on the uniprot_all_2021_04.fa (101GB) dataset of af3.

**Comment by milot-mirdita** [comment]
```
CUDA error: out of memory : /home/vsts/work/1/s/lib/libmarv/src/cudasw4.cuh, line 1276
```
This is a very odd place for a crash. How did you compile MMseqs2? Can you try with the precompiled binary instead please:
https://github.com/soedinglab/MMseqs2/releases/download/16-747c6/mmseqs-linux-gpu.tar.gz

**Comment by bestz123** [comment]
> This is a very odd place for a crash. How did you compile MMseqs2? Can you try with the precompiled binary instead please:
> https://github.com/soedinglab/MMseqs2/releases/download/16-747c6/mmseqs-linux-gpu.tar.gz

Yes, I am using this precompiled version, and this error sometimes does not appear when I re-run it.

**Comment by milot-mirdita** [comment]
Are the GPUs already in use by other processes? Can you try to explicitly set `CUDA_VISIBLE_DEVICES` to the GPU you want to use?

**Comment by milot-mirdita** [comment]
Also can you check if you previously started a MMseqs2 GPU server and didn't clean it up (`ps aux | grep gpuserver`)?

**Comment by bestz123** [comment]
> Are the GPUs already in use by other processes? Can you try to explicitly set `CUDA_VISIBLE_DEVICES` to the GPU you want to use?

I have checked that the GPU is not occupied by other tasks. I use CUDA_VISIBLE_DEVICES=0


> Also can you check if you previously started a MMseqs2 GPU server and didn't clean it up (`ps aux | grep gpuserver`)?

Before I start the MMseqs2 GPU server, the GPU memory usage is normal.

Could it be related to the length of my alignment sequence? The longer the sequence, the easier it is to burst the video memory. Or could it be related to the mode I set? When I choose --num-iterations 3, this error often occurs.
